public/js/monitor.js

```diff
@@ -36,19 +37,100 @@ function isValidEmail(val) {
 }
 
 
-function doOauth(el) {
+function doOauth(el, {emailWatch = false} = {}) {
+  // Growth Experiment: To sidestep the breach scans form, we have to check if
+  // there is an email address entered into #scan-user-email form.
+  // If so, we set it similiar to what would happen on form submit.
+  //
+  // Options was added to limit how the watched email input field is injected
+  // in this function. It only moves it if  options.emailWatch is set to TRUE;
+
   let url = new URL("/oauth/init", document.body.dataset.serverUrl);
   url = getFxaUtms(url);
+
   ["flowId", "flowBeginTime", "entrypoint"].forEach(key => {
     url.searchParams.append(key, encodeURIComponent(el.dataset[key]));
   });
+
+  if (!sessionStorage) {
+    window.location.assign(url);
+    return;
+  }
+
+  const scannedEmailId = document.querySelector("#scan-user-email input[name=scannedEmailId]");
+
+  // Preserve entire control function
+  if (!emailWatch) {
+    if (sessionStorage && sessionStorage.length > 0) {
+      const lastScannedEmail = sessionStorage.getItem(`scanned_${sessionStorage.length}`);
+      if (lastScannedEmail) {
+        url.searchParams.append("email", lastScannedEmail);
+      }
+    }
+    window.location.assign(url);
+    return;
+  }
+
+  // Growth Experiment: This logic is complex to handle the different scenarios of users logging into FxA.
+  let email = false;
+
+  let growthTaggedURL = new URL("/oauth/init", document.body.dataset.serverUrl);
+  growthTaggedURL = appendFxaParams(growthTaggedURL, document.body.dataset);
+
+  ["flowId", "flowBeginTime", "entrypoint", "form_type", "entrypoint_experiment", "entrypoint_variation"].forEach(key => {
```

Comment:
And then just append "entrypoing_experiment" and "entrypoint_variation" here?

Reply:
I added one qualifier – making sure there is a value to inject before placing the tag.